src/main/java/com/server/HTTPServerManager.java

```diff
@@ -7,38 +7,47 @@
 
     private ServerConfig serverConfig;
     private RequestRouter requestRouter;
+    private boolean logRequests;
 
     HTTPServerManager(ServerConfig serverConfig, RequestRouter requestRouter){
         this.serverConfig = serverConfig;
         this.requestRouter = requestRouter;
+        this.logRequests = false;
     }
 
-    public void runServer() throws IOException {
+    HTTPServerManager(ServerConfig serverConfig, RequestRouter requestRouter, boolean logRequests){
+        this.serverConfig = serverConfig;
+        this.requestRouter = requestRouter;
+        this.logRequests = logRequests;
+    }
 
+    public void runServer() throws IOException {
         while(running()) {
-
-            ServerSocket serverSocket = createServerSocket();
-            Socket clientSocket = openSocket(serverSocket);
-
-            BufferedReader in = openInputStream(clientSocket);
-
-            RequestReader requestHeaderReader = new RequestReader(in);
-            RequestParser requestParser = new RequestParser(requestHeaderReader.getRequest(), serverConfig.getDirectory());
-
-            RequestParams requestParams = requestParser.getRequestParams();
-
-            ResponseBuilder responseBuilder = new ResponseBuilder(requestRouter);
-
-            ByteArrayOutputStream baos = new ByteArrayOutputStream();
-            baos.write(responseBuilder.getResponse(requestParams));
-            byte[] result = baos.toByteArray();
-            BufferedOutputStream outputStream = new BufferedOutputStream(clientSocket.getOutputStream());
-            outputStream.write(result);
-            baos.flush();
-            baos.close();
-            outputStream.close();
-
-            stopServer(serverSocket, clientSocket);
+            ServerSocket serverSocket = null;
+            Socket clientSocket = null;
+            try {
+                serverSocket = createServerSocket();
+                clientSocket = openSocket(serverSocket);
+
+                BufferedReader in = openInputStream(clientSocket);
+
+                RequestReader requestReader = new RequestReader(in);
+                String header = requestReader.getRequest();
```

Comment:
Does `requestReader.getRequest()` return the entire request, i.e., all headers and body?